src/components/job-common.js

```diff
@@ -36,3 +60,46 @@ export const statusIcon = (status, style) => {
       return submittedIcon(style)
   }
 }
+
+export const makeStatusLine = (iconFn, text) => div({ style: { display: 'flex', marginTop: '0.5rem', fontSize: 14 } }, [
+  iconFn({ marginRight: '0.5rem' }), text
+])
+
+export const makeSection = (label, children) => div({
+  style: {
+    flex: '0 0 33%', padding: '0 0.5rem 0.5rem',
+    whiteSpace: 'pre', textOverflow: 'ellipsis', overflow: 'hidden'
+  }
+}, [
+  div({ style: Style.elements.sectionHeader }, label),
+  h(Fragment, children)
+])
+
+const breadcrumbHistoryCaret = icon('angle-right', { size: 10, style: { margin: '0 0.25rem' } })
+
+export const jobHistoryBreadcrumbPrefix = (namespace, workspaceName) => {
+  return h(Fragment, [
+    h(Link, {
+      href: Nav.getLink('workspace-job-history', { namespace, name: workspaceName })
+    }, [icon('arrowLeft', { style: { marginRight: '0.5rem' } }), 'Job History']),
+    breadcrumbHistoryCaret
+  ])
+}
+
+export const submissionDetailsBreadcrumbSubtitle = (namespace, workspaceName, submissionId) => {
+  return div({ style: { marginBottom: '1rem', display: 'flex', alignItems: 'center' } }, [
+    jobHistoryBreadcrumbPrefix(namespace, workspaceName),
+    div({ style: Style.elements.sectionHeader }, [`Submission ${submissionId}`])
+  ])
+}
+
+export const workflowDetailsBreadcrumbSubtitle = (namespace, workspaceName, submissionId, workflowId) => {
```

Comment:
TOL: this and the submission details one above are only used in one place, do we want to move them there?

Reply:
IMO the benefits of declaring them both in the same place here are:
* If one of them changes, it's more clear that the others should change too
* If we copy/paste one to generate another subtitle link, it might be easier to refactor out some commonality

I don't see a huge downside to switching over to this file to see the method declarations so I'm inclined to leave them here.

Reply:
fair enough!